geojson_modelica_translator/modelica/csv_modelica.py

```
@@ -27,53 +27,91 @@
-        # copy the first line
-        timeseries_15min= timeseries_output.loc[[0],:]
-        timeseries_output = pd.concat([timeseries_output,timeseries_15min]).sort_index()
+        self.timeseries_output = pd.read_csv(input_csv_file_path)
+        # round the data command since energyplus reports too many sigfigs.
+        self.timeseries_output = self.timeseries_output.round(2)
+        # copy the first line since Dymola wants to have time start at zero.
+        timeseries_15min = self.timeseries_output.loc[[0], :]
+        self.timeseries_output = pd.concat([self.timeseries_output, timeseries_15min]).sort_index()
         # reset index
-        timeseries_output = timeseries_output.reset_index(drop=True)
-        #Nominal massFlowRate
-        nominalHea_massFloRate=pd.DataFrame({'#heating':['#Nominal heating water mass flow rate'],'#value':    [timeseries_output['massFlowRateHeating'].max()]},
-                                                columns=['#heating','#value'])
-        nominalCoo_massFloRate=pd.DataFrame({'#cooling':['#Nominal chilled water mass flow rate'],'#value':[timeseries_output['massFlowRateCooling'].max()]},
-                                                columns=['#cooling','#value'])
-
-    def timeseries_to_modelica_data(file_path, EnergyPlustimestep, data_type, output_modelica_file_name):
-# evaluate dimensions of the matrix
-        size = timeseries_output.shape
-# modifiy the index for modelica mos
-        timeseries_output.index = (timeseries_output.index)*EnergyPlustimestep
-        timeseries_output.index.name = '#time'
-        timeseries_output.drop(['Date/Time'], axis=1, inplace=True)
-# write to csv for modelica
-        file = file_name + '.csv'
-        with open(file,'w') as f:
+        self.timeseries_output = self.timeseries_output.reset_index(drop=True)
+
+        # Note that the file that is in this test repo does not have the column you are looking for (massFlowRateHeating),
```

The mass_flow_rates file that you sent does not have the massFlowRateheating or massFlowRateCooling columns. I think this will work once those columns exist.

---

Thinking further... we should write an assertion in case the columns don't exist and let this fail nicely. Maybe work for another week.